packages/hoppscotch-app/helpers/curlparser.ts

```diff
@@ -2,105 +2,77 @@ import * as URL from "url"
-        ? joinDataArguments(parsedArguments["data-binary"])
-        : parsedArguments["data-binary"],
-      dataArray: Array.isArray(parsedArguments["data-binary"])
-        ? parsedArguments["data-binary"]
-        : null,
-      isDataBinary: true,
-    }
-  } else if (parsedArguments.d) {
-    return {
-      data: Array.isArray(parsedArguments.d)
-        ? joinDataArguments(parsedArguments.d)
-        : parsedArguments.d,
-      dataArray: Array.isArray(parsedArguments.d) ? parsedArguments.d : null,
-      isDataBinary: false,
-    }
-  } else if (parsedArguments["data-ascii"]) {
-    return {
-      data: Array.isArray(parsedArguments["data-ascii"])
-        ? joinDataArguments(parsedArguments["data-ascii"])
-        : parsedArguments["data-ascii"],
-      dataArray: Array.isArray(parsedArguments["data-ascii"])
-        ? parsedArguments["data-ascii"]
-        : null,
-      isDataBinary: false,
-    }
-  }
-}
+const joinDataArguments = (dataArguments: string[]) => dataArguments.join("&")
 
 const parseCurlCommand = (curlCommand: string) => {
-  const newlineFound = /\\/gi.test(curlCommand)
-  if (newlineFound) {
-    // remove '\' and newlines
-    curlCommand = curlCommand.replace(/\\/gi, "")
-    curlCommand = curlCommand.replace(/\n/g, "")
-  }
+  // remove '\' and newlines
+  curlCommand = curlCommand.replace(/ ?\\ ?$/gm, " ")
+  curlCommand = curlCommand.replace(/\n/g, "")
+
+  // remove all $ symbols from start of argument values
+  curlCommand = curlCommand.replaceAll("$'", "'")
+  curlCommand = curlCommand.replaceAll('$"', '"')
+
   // replace string for insomnia
-  curlCommand = curlCommand.replace(/--request /, "-X ")
-  curlCommand = curlCommand.replace(/--header /, "-H ")
-  curlCommand = curlCommand.replace(/--url /, " ")
-  curlCommand = curlCommand.replace(/-d /, "--data ")
+  const replaceables: { [key: string]: string } = {
+    "--request": "-X",
+    "--header": "-H",
+    "--url": "",
+    "--form": "-F",
+    "--data-raw": "--data",
+    "--data": "-d",
+    "--user": "-u",
+  }
+  for (const r in replaceables) {
+    for (const sym of ["'", '"', " "])
+      curlCommand = curlCommand.replace(RegExp(r + sym), replaceables[r] + sym)
+  }
 
   // yargs parses -XPOST as separate arguments. just prescreen for it.
-  curlCommand = curlCommand.replace(/ -XPOST/, " -X POST")
-  curlCommand = curlCommand.replace(/ -XGET/, " -X GET")
-  curlCommand = curlCommand.replace(/ -XPUT/, " -X PUT")
-  curlCommand = curlCommand.replace(/ -XPATCH/, " -X PATCH")
-  curlCommand = curlCommand.replace(/ -XDELETE/, " -X DELETE")
+  curlCommand = curlCommand.replace(/-XPOST/, " -X POST")
+  curlCommand = curlCommand.replace(/-XGET/, " -X GET")
+  curlCommand = curlCommand.replace(/-XPUT/, " -X PUT")
+  curlCommand = curlCommand.replace(/-XPATCH/, " -X PATCH")
+  curlCommand = curlCommand.replace(/-XDELETE/, " -X DELETE")
   curlCommand = curlCommand.trim()
   const parsedArguments = parser(curlCommand)
```

Comment:
This entire bit of code seems to be related to pre-processing the cURL command. Can we just break this entire bit off into a separate function outside ?

Reply:
You got it.